include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -0,0 +1,467 @@
+            case buffer_type::VIEWS:
+                // TODO: Implement
+                SPARROW_ASSERT(false, "Not implemented");
+                return 0;
+            case buffer_type::TYPE_IDS:
+                return static_cast<std::size_t>(length);
+        }
+        mpl::unreachable();
+    }
+
+    [[nodiscard]] std::vector<sparrow::buffer_view<uint8_t>> arrow_proxy::buffers() const
+    {
+        const auto buffer_count = static_cast<size_t>(array().n_buffers);
+        std::vector<sparrow::buffer_view<uint8_t>> buffers;
+        buffers.reserve(buffer_count);
+        const data_type data_type = format_to_data_type(schema().format);
+        const std::vector<buffer_type> buffers_type = get_buffer_types_from_data_type(data_type);
+        for (std::size_t i = 0; i < buffer_count; ++i)
+        {
+            const auto buffer_type = buffers_type[i];
+            auto buffer = array().buffers[i];
+            const std::size_t buffer_size = compute_buffer_size(buffer_type, length(), data_type);
+            auto* ptr = reinterpret_cast<uint8_t*>(const_cast<void*>(buffer));
+            buffers.emplace_back(ptr, buffer_size);
+        }
+        return buffers;
+    }
+
+    [[nodiscard]] std::vector<arrow_proxy> arrow_proxy::children() const
+    {
+        std::vector<arrow_proxy> children;
+        children.reserve(static_cast<std::size_t>(array().n_children));
+        for (int64_t i = 0; i < array().n_children; ++i)
+        {
+            children.emplace_back(array().children[i], schema().children[i]);
+        }
+        return children;
+    }
+
+    [[nodiscard]] std::optional<arrow_proxy> arrow_proxy::dictionary() const
+    {
+        if (array().dictionary == nullptr)
+        {
+            return std::nullopt;
+        }
+        return arrow_proxy{array().dictionary, schema().dictionary};
+    }
+
+    void arrow_proxy::release()
+    {
+        if (array().release != nullptr)
+        {
+            array().release(&array());
+        }
+        if (schema().release != nullptr)
+        {
+            schema().release(&schema());
+        }
+    }
+
+    [[nodiscard]] bool arrow_proxy::is_released() const
+    {
+        return array().release == nullptr && schema().release == nullptr;
+    }
+
+    [[nodiscard]] bool arrow_proxy::is_created_with_sparrow() const
+    {
+        return (array().release == &sparrow::release_arrow_array)
+               && (schema().release == &sparrow::release_arrow_schema);
+    }
+
+    [[nodiscard]] void* arrow_proxy::private_data() const
+    {
+        return array().private_data;
+    }
+
+    template <class... Ts>
+    struct overloaded : Ts...
+    {
+        using Ts::operator()...;
+    };
+    // Although not required in C++20, clang needs it to build the code below
+    template <class... Ts>
+    overloaded(Ts...) -> overloaded<Ts...>;
+
+    template <typename T>
+    [[nodiscard]] T& get_variant(auto& var)
```

Comment:
The name is confusing, and this function should probably be a private method or `array_proxy`.